stdpopsim/species.py

```diff
@@ -113,6 +115,77 @@ class Species(object):
     demographic_models = attr.ib(factory=list, kw_only=True)
     genetic_maps = attr.ib(factory=list, kw_only=True)
 
+    def get_chunk(self, length, chromosome=None, position=None, genetic_map=None):
+        """
+        Returns a :class:`.Contig` instance describing a chunk of the genome
+        to be simulated, based on empirical information for a given species
+        and chromosome.
+        A chunk will be randomly drawn from the genome if only the ``length``
+        parameter is provided. If ``chromosome`` is specified, the chunk is
+        drawn from that chromosome. A specific region of the genome may be
+        requested by specifying both the ``chromosome`` and ``position``
+        paramters.
+
+        :param int length: The length of the requested chunk.
+        :param str chromosome: The chromosome from which the chunk should be
+            drawn.
+        :param int position: The zero-based genomic coordinate to which the
+            chunk should correspond.
+        :param str genetic_map: If specified, obtain recombination rate information
+            from the genetic map with the specified ID. If None, simulate
+            using a default uniform recombination rate.
+        :rtype: :class:`.Contig`
+        :return: A :class:`.Contig` describing a simulation of the section of genome.
+        """
+        if chromosome is None and position is not None:
+            raise ValueError(f"Position {position} provided without chromosome.")
+        if type(length) != int or length <= 0:
+            raise ValueError(f"Invalid length {length}.")
+
+        if chromosome is not None:
+            chrom = self.genome.get_chromosome(chromosome)
+            if length > chrom.length:
+                raise ValueError(f"{chrom.id} not long enough for length {length}")
+        else:
+            w = []
+            for i, ch in enumerate(self.genome.chromosomes):
+                wl = ch.length if ch.length >= length else 0
+                w.append(wl if i == 0 else wl + w[i-1])
+            if w[-1] == 0:
+                raise ValueError(f"No chromosomes long enough for length {length}")
+            chrom = random.choices(self.genome.chromosomes, cum_weights=w)[0]
+
+        if position is None:
+            position = random.randrange(chrom.length - length)
+
+        if genetic_map is None:
+            gm = None
+            recomb_map = msprime.RecombinationMap.uniform_map(
+                    length, chrom.recombination_rate)
+        else:
+            # Extract a subset of the full genetic map.
```

Comment:
Let's move this basic functionality to msprime: https://github.com/tskit-dev/msprime/issues/901

Reply:
Ok.